test/Convex.test.ts

```diff
@@ -0,0 +1,402 @@
+          const mockCozyToken = <MockCozyToken>await deployContract(deployer, mockCozyTokenArtifact, [trigger.address]);
+          expect(await mockCozyToken.isTriggered()).to.be.false;
+
+          await setCrvTotalSupply(MAX_UINT, poolType as 'base' | 'meta');
+          await mockCozyToken.checkAndToggleTrigger();
+          expect(await mockCozyToken.isTriggered()).to.be.true;
+        });
+
+        it(`properly accounts for virtual price tolerance: curve ${poolType} pool`, async () => {
+          // Modify the currently stored virtual price by a set tolerance
+          async function modifyLastVirtualPrice(numerator: bigint, denominator: bigint) {
+            const isBase = poolType === 'base';
+            const crvPool = isBase ? crvBase : crvMeta;
+            const crvToken = isBase ? crvBaseToken : crvMetaToken;
+            const vpMethod = isBase ? 'lastVpBasePool' : 'lastVpMetaPool';
+
+            const lastVirtualPrice = (await trigger[vpMethod]()).toBigInt();
+            const lastTotalSupply = (await crvToken.totalSupply()).toBigInt();
+            const newTotalSupply = (lastTotalSupply * denominator) / numerator;
+            const newVirtualPrice = (lastVirtualPrice * numerator) / denominator;
+            await setCrvTotalSupply(newTotalSupply, poolType as 'base' | 'meta');
+            // Due to rounding error, values may sometimes differ by 1 wei, so validate with a tolerance +/2 wei
+            expect(await crvPool.get_virtual_price()).to.be.above(newVirtualPrice - 2n);
+            expect(await crvPool.get_virtual_price()).to.be.below(newVirtualPrice + 2n);
+          }
+
+          // Read the trigger's tolerance
+          const tolerance = (await trigger.virtualPriceTol()).toBigInt();
+
+          // Increase virtual price to a larger value, should NOT be triggered (sanity check)
+          await modifyLastVirtualPrice(101n, 100n); // 1% increase
+          await assertTriggerStatus(false);
+
+          // Decrease virtual price by an amount less than tolerance, should NOT be triggered
+          await modifyLastVirtualPrice(99n, 100n); // 1% decrease
+          await assertTriggerStatus(false);
+
+          // Decrease virtual price by an amount exactly equal to tolerance, should NOT be triggered
+          await modifyLastVirtualPrice(tolerance, 1000n);
+          await assertTriggerStatus(false);
+
+          // Decrease virtual price by an amount more than tolerance, should be triggered
+          await modifyLastVirtualPrice(tolerance - 1n, 1000n);
+          await assertTriggerStatus(true);
+        });
+
+        // Due to a bug in Smock (https://github.com/defi-wonderland/smock/issues/101) we leave these tests
+        // skipped because they cause other tests to fail. Replace the `.skip` with a `.only` to run these
+        // tests and verify that they pass
+        it.skip(`toggles trigger when ${poolType} pool's get_virtual_price() reverts`, async () => {
```

Comment:
Random thought: if you manually called setupFixture here, effectively deploying a new set of contracts for use only with this test, would it work around this issue allowing the test to be included in the suite without skipping?

Reply:
Ah good idea. It sounds like it should work, but in my quick tests it did not, so leaving this as-is for now in the interest of time 😕